src/core_plugins/timelion/server/series_functions/es/lib/create_date_agg.js

```diff
@@ -34,6 +34,11 @@ export default function createDateAgg(config, tlConfig, scriptedFields) {
       const metricName = metric[0] + '(' + metric[1] + ')';
       dateAgg.time_buckets.aggs[metricName] = {};
       dateAgg.time_buckets.aggs[metricName][metric[0]] = buildAggBody(metric[1], scriptedFields);
+      if (metric[0] === 'percentiles' && metric[2]) {
+        let percentList = metric[2].split(',');
+        percentList = percentList.map(x => parseFloat(x));
+        dateAgg.time_buckets.aggs[metricName][metric[0]].percents = percentList;
```

Comment:
This needs to use the buildAggBody method so scripted fields can be supported.

Reply:
Scripted fields are already supported. 

```
36: dateAgg.time_buckets.aggs[metricName][metric[0]] = buildAggBody(metric[1], scriptedFields);
```

`buildAggBody` sets `dateAgg.time_buckets.aggs[metricName][metric[0]]` to either `field` or `script`, `percents` is added to the corresponding object afterwards.

Reply:
you are right. disregard